source/common/access_log/access_log_impl.cc

```diff
@@ -284,5 +285,8 @@ AccessLogFactory::fromProto(const envoy::config::filter::accesslog::v2::AccessLo
   return factory.createAccessLogInstance(*message, std::move(filter), context);
 }
 
+REGISTER_EXTENSION_FACTORY(Server::Configuration::AccessLogInstanceFactory, "access_loggers");
```

Comment:
The way I had been imagining this to work is:
1) Every factory type has some type of built-in name (so built into `Server::Configuration::AccessLogInstanceFactory`)
2) During the constructor/registration of an individual factory, it would register itself into the factory-wide map.

I think if we do that we don't need any extra registration code like this? WDYT?

Reply:
Switched over to automatic registration.